internal/proto/controller/api/resources/hostcatalogs/v1/host_catalog.proto

```diff
@@ -14,50 +14,52 @@ import "controller/custom_options/v1/options.proto";
+	string id = 10;  // @gotags: `class:"public"`
 
 	// The ID of the Scope of which this Host Catalog is a part.
-	string scope_id = 20 [json_name="scope_id"];
+	string scope_id = 20 [json_name="scope_id"];  // @gotags: `class:"public"`
 
 	// Output only. Scope information for this resource.
 	resources.scopes.v1.ScopeInfo scope = 30;
 
 	// The ID of the plugin of which this catalog is created.
-	string plugin_id = 34 [json_name="plugin_id", (custom_options.v1.generate_sdk_option) = true];
+	string plugin_id = 34 [json_name="plugin_id", (custom_options.v1.generate_sdk_option) = true];  // @gotags: `class:"public"`
 
 	// Output only. Plugin information for this resource.
-	resources.plugins.v1.PluginInfo plugin = 35;
+	resources.plugins.v1.PluginInfo plugin = 35;  // @gotags: `class:"public"`
 
 	// Optional name for identification purposes.
-	google.protobuf.StringValue name = 40 [(custom_options.v1.generate_sdk_option) = true, (custom_options.v1.mask_mapping) = {this:"name" that: "name"}];
+	google.protobuf.StringValue name = 40 [(custom_options.v1.generate_sdk_option) = true, (custom_options.v1.mask_mapping) = {this:"name" that: "name"}];  // @gotags: `class:"public"`
 
 	// Optional user-set description for identification purposes.
-	google.protobuf.StringValue description = 50 [(custom_options.v1.generate_sdk_option) = true, (custom_options.v1.mask_mapping) = {this:"description" that: "description"}];
+	google.protobuf.StringValue description = 50 [(custom_options.v1.generate_sdk_option) = true, (custom_options.v1.mask_mapping) = {this:"description" that: "description"}];  // @gotags: `class:"public"`
 
 	// Output only. The time this resource was created.
-	google.protobuf.Timestamp created_time = 60 [json_name="created_time"];
+	google.protobuf.Timestamp created_time = 60 [json_name="created_time"];  // @gotags: `class:"public"`
 
 	// Output only. The time this resource was last updated.
-	google.protobuf.Timestamp updated_time = 70 [json_name="updated_time"];
+	google.protobuf.Timestamp updated_time = 70 [json_name="updated_time"];  // @gotags: `class:"public"`
 
 	// Version is used in mutation requests, after the initial creation, to ensure this resource has not changed.
 	// The mutation will fail if the version does not match the latest known good version.
-	uint32 version = 80;
+	uint32 version = 80;  // @gotags: `class:"public"`
 
 	// The type of Host Catalog.
-	string type = 90;
+	string type = 90;  // @gotags: `class:"public"`
 
-	// Attributes specific to the catalog type.
-	google.protobuf.Struct attributes = 100 [(custom_options.v1.generate_sdk_option) = true];
+	oneof attrs {
+		// Attributes specific to the catalog type.
+		google.protobuf.Struct attributes = 100 [(custom_options.v1.generate_sdk_option) = true, (custom_options.v1.subtype) = "default"];
+	}
 
 	// Secrets specific to the catalog type. These are never output.
 	google.protobuf.Struct secrets = 110 [(custom_options.v1.generate_sdk_option) = true];
```

Comment:
This is related to my earlier comment/question about google proto types. `secrets` needs to be redacted and isn't marked here. If it will be done automatically with default handling of those types then no worries. It might be better to mark them explicitly though. I also don't know the current state of redacting maps...

Reply:
It currently only redacts the values, not the keys: https://github.com/hashicorp/boundary/pull/2004/files#diff-158f011052a662828d487cf6d4fd1f971593ebe611ae7dd6658126ecc100e0a3R124.